* Functional impact of SNIPS in pathway context?
    * Map proteins to genes
    * Identify some SNIPs that are in regulatory regions (look at annotations)
    * Associate SNIPS with function 
    * analyse them with RegulomeDB 
        * Use RegulomeDB to identify DNA features and regulatory elements in non-coding regions of the human genome by entering
* WikiPathways
    * Browse different pathways
    * Everyone can edit them (like wikipedia)
    * Create Pathway cartoons to make the pathway knowledge more readable
    * Editor creates computer-readable format for pathway
    * more data than KEGG
    * Some data is badly annotated and should only be used with car
* Open Science:
    * Free knowledge for everyone without paywalls 
    * Open Access + Open Source + Open collaboration
    * Solves Reproducible Crisis (80% of current papers are not reproducible, because samples, parameters or other data are not shared)
    * Higher Quantity / Quality 
    * Less stressful for publication
* Problems with Pathways
    * Lot of data is useless because it has no computer readable format (1000 static figures per month)
    * Scaling and sustainability
    * Changing the current mindset of pathway scientists
    * different databases have different data formats
    * Representation (sketches) is often too linear and does not show the complete scope
    * There is no general standard
* Example for pathway database usage
    * healthy <-> diseases
    * Blastula <-> Dorso / ventral polarity
    * How did tumor cells lose their growth control?
    * Search for pathways including your studied / found gene expressions
* GSEA (Gen Set Enrichment Analysis)
    * Statistical approach to match data pattern to pathway
    * Gene set enrichment analysis (GSEA) (also functional enrichment analysis) is a method to identify classes of genes/ proteins that are over-represented in a large set of genes/proteins, and may have an association with disease phenotypes. The method uses statistical approaches to identify significantly enriched/depleted groups of genes.
* KEGG Pathway Database
	* is curated
    * Orthology: homologous sequences descended from the same ancestral sequence separated by a speciation event.
    * Doing normalization on data
    * Ortholog table: Map genes to pathways
    * KEGG Orthology: Find similar pathways in other organisms
    * tries to take genetic variation in account (hard to estimate)
    * Disease focus
